compiler-rt/src/alu/sshl_sat.cairo

```diff
@@ -0,0 +1,88 @@
+pub mod sshl_sat_i8;
+
+use crate::utils::{assert_fits_in_type, extend_sign};
+use crate::alu::shl::shl;
+use core::num::traits::{BitSize, Bounded};
+
+// Perform the `sshl_sat` operation.
+//
+// This is a generic implementation for every data type. Its specialized versions
+// are defined and tested in the sshl_sat/sshl_sat_<type>.cairo files.
+fn sshl_sat<
```

Comment:
I want to see more explanation of the actual operation here, given that you have figured it out based on test cases from LLVM's `compiler-rt`. I am not saying we want to just comment the implemented algorithm as that is pretty self-explanatory, but I want to see the _requirements_ on the operation (particularly its edge-case behavior) explained here.